crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -644,22 +645,19 @@ impl<'a> Resolver<'a> {
         }
     }
 
-    fn handle_contract_visibility(&mut self, func: &NoirFunction) -> Option<ContractVisibility> {
-        let mut contract_visibility = func.def.contract_visibility;
-
-        if self.in_contract() && contract_visibility.is_none() {
-            // The default visibility is 'secret' for contract functions without visibility modifiers
-            contract_visibility = Some(ContractVisibility::Secret);
-        }
-
-        if !self.in_contract() && contract_visibility.is_some() {
-            contract_visibility = None;
-            self.push_err(ResolverError::ContractVisibilityInNormalFunction {
-                span: func.name_ident().span(),
-            })
+    fn handle_contract_visibility(&mut self, func: &NoirFunction) -> Option<ContractFunctionType> {
```

Comment:
Should be renamed to `handle_contract_function_type`?

Reply:
All instances should be fixed now